sound/soc/sof/intel/hda.c

```diff
@@ -414,6 +415,21 @@ void hda_dsp_dump_skl(struct snd_sof_dev *sdev, u32 flags)
 	}
 }
 
+/* dump the first 8 dwords representing the extended ROM status */
+static void hda_dsp_dump_ext_rom_status(struct snd_sof_dev *sdev)
+{
+	char msg[128];
+	u32 value;
+	int i;
+
+	for (i = 0; i < HDA_EXT_ROM_STATUS_SIZE; i++) {
+		value = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_SRAM_REG_ROM_STATUS + i * 0x4);
+		snprintf(msg + strlen(msg), sizeof(msg) - strlen(msg), "0x%x ", value);
```

Comment:
@ranj063 Hmm, msg is not necessarily null initialized, so the first strlen can overrun the buffer. I'd use a separate length variable.. it's simple and gets rid of repeated calls to strlen.

Reply:
@kv2019i ack. done